easybuild/framework/easyblock.py

```diff
@@ -1734,8 +1737,16 @@ def prepare_step(self, start_dir=True):
         if not self.build_in_installdir:
             self.rpath_filter_dirs.append(self.builddir)
 
+        # always include self.installdir+'/lib' and self.installdir+'/lib64' and $ORIGIN
+        # $ORIGIN will be resolved by the loader to be the full path to the executable or shared object
+        # see also https://linux.die.net/man/8/ld-linux;
+        self.rpath_include_dirs.append(self.installdir+'/lib')
+        self.rpath_include_dirs.append(self.installdir+'/lib64')
+        self.rpath_include_dirs.append('$ORIGIN')
```

Comment:
@jdonners Shouldn't we also retain `$ORIGIN/../lib` and `$ORIGIN/../lib64` in here? That's not necessarily the same as `<installdir>/lib` and `<installdir>/lib64`, is it?

Reply:
@boegel indeed they're not necessarily the same. Conservatively, I thought it's better to add fewer paths to the `RPATH`. I'd expect that packages that install shared objects in non-standard locations, also somehow make these directories findable in some wrapper scripts. There's no guarantee that that is indeed the case. On the other side, it still is an experimental feature..

Reply:
For now, I'd stick to also included `$ORIGIIN/../lib` and `$ORIGIN/../lib64`.

It indeed makes sense to limit the amount of paths we inject, but 3 or 5 isn't going to make a big difference I think.

The order may be important though (w.r.t. speeding up the loader): we should probably list the most common locations first?

Reply:
@jdonners Fixed in https://github.com/sara-nl/easybuild-framework/pull/1